src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -146,6 +146,9 @@ public AsymmetricSignatureProvider(SecurityKey key, string algorithm, bool willC
             if (asymmetricAlgorithmResolver != null)
             {
                 AsymmetricAlgorithm asymmetricAlgorithm = asymmetricAlgorithmResolver(key, algorithm, willCreateSignatures);
+                if (asymmetricAlgorithm == null)
+                    throw LogHelper.LogException<ArgumentOutOfRangeException>(LogMessages.IDX10646, nameof(key), algorithm);
```

Comment:
or InvalidOperationException. Either is fine. Point is the message will help the dev understand what the error is about.